WordPress Playground tests skip Node setup on push runs

## Problem

`Extra-Chill/homeboy-action@v2` can pass on `pull_request` runs but fail on the matching `push` run for the same workflow when the component uses the WordPress extension's Playground-backed test runner.

The BFB workflow added:

```yaml
- uses: Extra-Chill/homeboy-action@v2
  with:
    commands: test
    php-version: ${{ matrix.php-version }}
    autofix: 'false'
```

The PR run passed, but every subsequent `push` run on `main` failed before PHPUnit started.

## Repro / observed runs

Repo: https://github.com/chubes4/block-format-bridge

- Passing PR run: https://github.com/chubes4/block-format-bridge/actions/runs/24969016534
- Failing push run from the merge commit: https://github.com/chubes4/block-format-bridge/actions/runs/24969030485
- Example failed job: https://github.com/chubes4/block-format-bridge/actions/runs/24969030485/job/73108702322

The failed push run resolves:

```text
SCOPE_CONTEXT: push
SCOPE_MODE: full
PORTABLE_NODE:
Runtime env detected:
  php:  8.2 (overridden by input)
  node: skip
```

Then `homeboy test` fails before tests run:

```text
Error: @wp-playground/cli not found at /home/runner/.config/homeboy/extensions/wordpress/node_modules/.bin/wp-playground-cli

Install it with: cd /home/runner/.config/homeboy/extensions/wordpress && npm install
Or switch backends: homeboy component set <id> test_backend host

============================================
BUILD FAILED: Playground CLI setup
============================================
```

## Expected behavior

If the selected Homeboy extension/test backend requires Node tooling, `homeboy-action` should install/setup Node and extension npm dependencies even when the project under test has no repo-local Node metadata.

The WordPress extension's Playground backend depends on `@wp-playground/cli`, so the action should not infer `node: skip` solely from the target repo shape when `commands: test` will route through that backend.

## Actual behavior

On `push` runs, Node setup is skipped and the WordPress extension is installed without its npm dependencies. The test phase then fails as an infrastructure bootstrap failure, not as a project test failure.

## Secondary failure path

When the push-context test fails, the action attempts issue filing. Without sufficient permissions this can add a second failure path / confusing exit. BFB is currently adding:

```yaml
permissions:
  issues: write
```

But issue filing failure should not obscure the original bootstrap failure. If issue filing cannot run, the action should report that as a non-primary warning and preserve the original failure as the main signal.

## Workaround in consumer workflows

BFB is adding this repo-side workaround:

```yaml
permissions:
  issues: write

with:
  node-version: '24'
```

That should unblock BFB, but it is not a good default requirement for every WordPress-extension consumer. The action/extension already knows the selected extension and command; the consumer should not need to know that WordPress Playground requires Node.

## Acceptance criteria

- WordPress-extension Playground tests install `@wp-playground/cli` on both `pull_request` and `push` runs without requiring each consumer workflow to pass `node-version` explicitly.
- Runtime detection accounts for extension/backend requirements, not only repo-local package metadata.
- If categorized issue filing fails because permissions are missing or GitHub rejects the request, the action reports a warning and preserves the original command failure as the primary failure.

## AI assistance

- **AI assistance:** Yes
- **Tool(s):** OpenCode (GPT-5.5)
- **Used for:** Diagnosed the BFB push-context CI failure and drafted this upstream issue; Chris remains responsible for triage and implementation.

--- a/action.yml
+++ b/action.yml
@@ -308,6 +308,7 @@ runs:
         PHP_INPUT: ${{ inputs.php-version }}
         NODE_INPUT: ${{ inputs.node-version }}
         COMPONENT_DIR: ${{ steps.read-config.outputs.component-dir }}
+        PORTABLE_EXTENSION: ${{ steps.read-config.outputs.portable-extension }}
       run: bash ${{ github.action_path }}/scripts/setup/detect-runtime-env.sh
 
     - name: Setup PHP
@@ -546,6 +547,7 @@ runs:
         && steps.categorized-issues.outcome != 'success'
         && (inputs.auto-issue == 'true' || (inputs.auto-issue == '' && github.event_name != 'pull_request'))
       shell: bash
+      continue-on-error: true
       env:
         GH_TOKEN: ${{ inputs.app-token || github.token }}
         RESULTS: ${{ steps.select-results.outputs.results }}
@@ -554,4 +556,8 @@ runs:
         BINARY_SOURCE: ${{ steps.resolve-binary.outputs.binary-source }}
         AUTOFIX_ATTEMPTED: ${{ steps.autofix-prepare-nonpr.outputs.committed }}
         AUTOFIX_PR_CREATED: ${{ steps.autofix-open-pr.outputs.created }}
-      run: bash ${{ github.action_path }}/scripts/issues/auto-file-issue.sh
+      run: |
+        bash ${{ github.action_path }}/scripts/issues/auto-file-issue.sh || {
+          echo "::warning::Auto-issue filing failed; preserving the original Homeboy command failure as the primary signal"
+          exit 0
+        }

--- a/scripts/setup/auto-setup-environment.sh
+++ b/scripts/setup/auto-setup-environment.sh
@@ -34,6 +34,13 @@ if [ -f "composer.json" ]; then
   fi
 fi
 
+# ── Extension setup ──
+
+if [ -n "${EXTENSION}" ] && command -v homeboy &>/dev/null; then
+  echo "Setting up Homeboy extension dependencies (${EXTENSION})..."
+  homeboy extension setup "${EXTENSION}" 2>&1
+fi
+
 # ── npm install (Node projects) ──
 
 if [ -f "package.json" ] && [ -n "${PORTABLE_NODE:-}" ]; then

--- a/scripts/setup/detect-runtime-env.sh
+++ b/scripts/setup/detect-runtime-env.sh
@@ -7,7 +7,9 @@
 # file header instead of requiring manual configuration.
 #
 # Action input overrides (PHP_INPUT, NODE_INPUT) take priority over
-# detected values.
+# detected values. Extension runtime requirements are used as a fallback when
+# the component itself does not declare a runtime but the installed extension
+# needs one for setup or execution.
 #
 # Outputs (GITHUB_ENV + GITHUB_OUTPUT):
 #   PORTABLE_PHP  — PHP version to install
@@ -18,6 +20,8 @@ set -euo pipefail
 PHP_INPUT="${PHP_INPUT:-}"
 NODE_INPUT="${NODE_INPUT:-}"
 COMPONENT_DIR="${COMPONENT_DIR:-.}"
+PORTABLE_EXTENSION="${PORTABLE_EXTENSION:-}"
+DEFAULT_EXTENSION_NODE_VERSION="${DEFAULT_EXTENSION_NODE_VERSION:-24}"
 
 PORTABLE_PHP=""
 PORTABLE_NODE=""
@@ -36,6 +40,15 @@ else
   DETECTED_NODE=""
 fi
 
+EXTENSION_NODE_REQUIRED="false"
+if [ -n "${PORTABLE_EXTENSION}" ] && command -v homeboy &>/dev/null; then
+  EXTENSION_JSON="$(homeboy extension show "${PORTABLE_EXTENSION}" 2>/dev/null || true)"
+  EXTENSION_PATH="$(echo "${EXTENSION_JSON}" | jq -r '.data.extension.path // empty' 2>/dev/null || true)"
+  if [ -n "${EXTENSION_PATH}" ] && [ -f "${EXTENSION_PATH}/package.json" ]; then
+    EXTENSION_NODE_REQUIRED="true"
+  fi
+fi
+
 # Action input overrides take priority
 if [ -n "${PHP_INPUT}" ]; then
   PORTABLE_PHP="${PHP_INPUT}"
@@ -47,6 +60,8 @@ if [ -n "${NODE_INPUT}" ]; then
   PORTABLE_NODE="${NODE_INPUT}"
 elif [ -n "${DETECTED_NODE}" ]; then
   PORTABLE_NODE="${DETECTED_NODE}"
+elif [ "${EXTENSION_NODE_REQUIRED}" = "true" ]; then
+  PORTABLE_NODE="${DEFAULT_EXTENSION_NODE_VERSION}"
 fi
 
 echo "PORTABLE_PHP=${PORTABLE_PHP}" >> "${GITHUB_ENV}"
@@ -57,4 +72,11 @@ echo "portable-node=${PORTABLE_NODE}" >> "${GITHUB_OUTPUT}"
 
 echo "Runtime env detected:"
 echo "  php:  ${PORTABLE_PHP:-skip}${PHP_INPUT:+ (overridden by input)}"
-echo "  node: ${PORTABLE_NODE:-skip}${NODE_INPUT:+ (overridden by input)}"
+if [ -n "${NODE_INPUT}" ]; then
+  NODE_NOTE=" (overridden by input)"
+elif [ -z "${DETECTED_NODE}" ] && [ "${EXTENSION_NODE_REQUIRED}" = "true" ]; then
+  NODE_NOTE=" (required by ${PORTABLE_EXTENSION} extension setup)"
+else
+  NODE_NOTE=""
+fi
+echo "  node: ${PORTABLE_NODE:-skip}${NODE_NOTE}"

--- a/scripts/setup/test-detect-runtime-env.sh
+++ b/scripts/setup/test-detect-runtime-env.sh
@@ -1,0 +1,81 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+assert_equals() {
+  local expected="$1"
+  local actual="$2"
+  local label="$3"
+
+  if [ "${expected}" != "${actual}" ]; then
+    printf 'FAIL: %s\nexpected: %s\nactual:   %s\n' "${label}" "${expected}" "${actual}"
+    exit 1
+  fi
+
+  printf 'PASS: %s\n' "${label}"
+}
+
+TMPDIR="$(mktemp -d)"
+trap 'rm -rf "${TMPDIR}"' EXIT
+
+FAKE_BIN="${TMPDIR}/bin"
+FAKE_EXTENSION="${TMPDIR}/extensions/wordpress"
+mkdir -p "${FAKE_BIN}" "${FAKE_EXTENSION}"
+printf '{"engines":{"node":">=18.12.0"}}\n' > "${FAKE_EXTENSION}/package.json"
+
+cat > "${FAKE_BIN}/homeboy" <<'SH'
+#!/usr/bin/env bash
+set -euo pipefail
+
+if [ "$1" = "component" ] && [ "$2" = "env" ]; then
+  printf '{"success":true,"data":{"entity":{"php":"8.2"}}}\n'
+  exit 0
+fi
+
+if [ "$1" = "extension" ] && [ "$2" = "show" ]; then
+  printf '{"success":true,"data":{"extension":{"path":"%s"}}}\n' "${FAKE_EXTENSION_PATH}"
+  exit 0
+fi
+
+printf 'unexpected fake homeboy invocation: %s\n' "$*" >&2
+exit 1
+SH
+chmod +x "${FAKE_BIN}/homeboy"
+
+GITHUB_ENV_FILE="${TMPDIR}/env"
+GITHUB_OUTPUT_FILE="${TMPDIR}/output"
+LOG_FILE="${TMPDIR}/log"
+
+PATH="${FAKE_BIN}:${PATH}" \
+FAKE_EXTENSION_PATH="${FAKE_EXTENSION}" \
+GITHUB_ENV="${GITHUB_ENV_FILE}" \
+GITHUB_OUTPUT="${GITHUB_OUTPUT_FILE}" \
+COMPONENT_DIR="${TMPDIR}" \
+PORTABLE_EXTENSION="wordpress" \
+DEFAULT_EXTENSION_NODE_VERSION="24" \
+bash "${SCRIPT_DIR}/detect-runtime-env.sh" > "${LOG_FILE}"
+
+assert_equals "PORTABLE_PHP=8.2" "$(grep '^PORTABLE_PHP=' "${GITHUB_ENV_FILE}")" "detects component PHP"
+assert_equals "PORTABLE_NODE=24" "$(grep '^PORTABLE_NODE=' "${GITHUB_ENV_FILE}")" "uses extension-required Node fallback"
+assert_equals "portable-node=24" "$(grep '^portable-node=' "${GITHUB_OUTPUT_FILE}")" "writes portable-node output"
+
+if ! grep -q 'node: 24 (required by wordpress extension setup)' "${LOG_FILE}"; then
+  printf 'FAIL: extension Node requirement is explained in log\n'
+  exit 1
+fi
+printf 'PASS: extension Node requirement is explained in log\n'
+
+GITHUB_ENV_FILE="${TMPDIR}/env-input"
+GITHUB_OUTPUT_FILE="${TMPDIR}/output-input"
+PATH="${FAKE_BIN}:${PATH}" \
+FAKE_EXTENSION_PATH="${FAKE_EXTENSION}" \
+GITHUB_ENV="${GITHUB_ENV_FILE}" \
+GITHUB_OUTPUT="${GITHUB_OUTPUT_FILE}" \
+COMPONENT_DIR="${TMPDIR}" \
+PORTABLE_EXTENSION="wordpress" \
+NODE_INPUT="20" \
+bash "${SCRIPT_DIR}/detect-runtime-env.sh" > /dev/null
+
+assert_equals "PORTABLE_NODE=20" "$(grep '^PORTABLE_NODE=' "${GITHUB_ENV_FILE}")" "node input overrides extension fallback"

